src/main/java/uk/gov/companieshouse/api/accounts/validation/ValidRptTransactionTypeImpl.java

```diff
@@ -0,0 +1,18 @@
+package uk.gov.companieshouse.api.accounts.validation;
+
+import org.apache.commons.lang.StringUtils;
+import org.springframework.stereotype.Component;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+@Component
+public class ValidRptTransactionTypeImpl implements ConstraintValidator<ValidRptTransactionType, String> {
+
+    @Override
+    public boolean isValid(String rptTransactionType, ConstraintValidatorContext context) {
+        return StringUtils.isBlank(rptTransactionType)
+                || rptTransactionType.trim().equalsIgnoreCase("money given to the company by a related party")
```

Comment:
Do we want to use ignorecase here?
What do we do elsewhere?
I only ask because these are machine-supplied values, not user-input. In a public facing API, I would make these enums with exact values. All of which means that you could tightly specify the string, including its case.

Reply:
Regarding trim...
Do you think that we should trim user input and store the trimmed form?

Reply:
We are trimming and are allowing when it comes to specifying the string in mixed case in for example FRS101 and FRS102 case. I remember, you i and Brad had discussion about this regarding FRS101 and FRS102 and we came up with trimming and ignoring cases when it comes to compare the user input with our values 🙂 

I am more than happy to use enums and restrict the user from providing exact value but it might make it a bit more difficult in case they have a habit of typing capitals for example

Reply:
The user can't provide an exact value can they? The text is from the radio button which we're supplying so we might be ok with enums with this?

Reply:
@mreis-ch This will be for API filers who use for example Postman to file their accounts. For web filers the value will be same as one of the selected radios